ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/FinalBlockConfirmation.java

```diff
@@ -44,7 +44,8 @@ static FinalBlockConfirmation genesisConfirmation(final MutableBlockchain blockc
   }
 
   static FinalBlockConfirmation ancestorConfirmation(final MutableBlockchain blockchain) {
-    return firstHeader -> blockchain.contains(firstHeader.getParentHash());
+    return firstHeader ->
+        blockchain.getChainHeadHeader().getHash().equals(firstHeader.getParentHash());
```

Comment:
This might this break backward sync for a reorg.    e.g. blockhain.contains() will return true if we have block A' but it isn't the chain head.  This check requires it to be the chain head...

Reply:
I'm using the blck number. Should be fine now but. if you can verify

Reply:
when I pull this branch here I see:
 `    return firstHeader -> blockchain.getChainHeadBlockNumber() + 1 >= firstHeader.getNumber();
`
I might be missing something, but I read the firstAncestor block as being the earliest block fetched by backward sync.  and finalBlockConfirmation.ancestorHeaderReached as the termination condition for fetching further back in BackwardSyncAlgorithm.pickNextStep.

If we are backward syncing because of a fork from 5 or 10 blocks behind head, I think this condition could terminate early.  it is almost logically identical to the conditional above it:
`    return firstHeader -> blockchain.getChainHeadBlockNumber() + 1 >= firstHeader.getNumber();
`

The first condition makes sense to me b/c we really just need to ensure this backward path resolves to something we have, rather than a particular height.

Reply:
Ok I think I understand your remark. On the other hand, I have to keep this check because it allows me to have the backardsync works after the heal in case we répond. Because we will have the blocks but I still want to go back to import them again . So I think I should add the initial condition + this one. In the case of a REORG the first condition will make sure that it does not stop too soon

Reply:
I proposed another fix on this PR https://github.com/hyperledger/besu/pull/5059/files#diff-0361d2ff12cd43f507bf88a3cb4d781a537ca765329e0811f60a9507527cbe13R47
Thanks